File: 프로그래머스/lv1/86491. 최소직사각형/최소직사각형.js
function solution(sizes) {
    let w = 0
    let h = 0
    
    for (let size of sizes) {
        if (size[0] > size[1]) {
            w = Math.max(w, size[0])
            h = Math.max(h, size[1])
        } else {
            w = Math.max(w, size[1])
            h = Math.max(h, size[0])
        }
    }
    
    return w * h
}

File: 프로그래머스/lv1/86491. 최소직사각형/최소직사각형.py
def solution(sizes):
    # 방법 2
    return max([max(size) for size in sizes]) * max([min(size) for size in sizes])

#     # 방법 1
#     w, h = 0, 0

#     for size in sizes:
#         long, short = (size[0], size[1]) if size[0] > size[1] else (size[1], size[0])
#         w = max(w, long)
#         h = max(h, short)
    
#     return w * h